src/test/lib/optimizer/strategy/predicate_placement_rule_test.cpp

```diff
@@ -629,6 +629,47 @@ TEST_F(PredicatePlacementRuleTest, HandleBarrierPredicatePushdown) {
   EXPECT_EQ(temporary_node->left_input(), inner_join);
 }
 
+TEST_F(PredicatePlacementRuleTest, HandleMultiPredicateBarrierPredicatePushdown) {
+  // In this test, the PredicatePlacementRule cannot push down the top two predicates because of a predicate having
+  // multiple outputs (the barrier node). In constrast to HandleBarrierPredicatePushdown, the barrier now is a
+  // multi-predicate semi-join (which should not be pushed down).
+
+  // clang-format off
+  const auto barrier_predicate_node =
+  JoinNode::make(JoinMode::Semi, expression_vector(equals_(_a_a, _b_a), not_equals_(_a_b, _b_b)),  // <-- 1st Predicate
+    PredicateNode::make(greater_than_(_b_b, 123),                                                  // <-- 2nd Predicate
+      JoinNode::make(JoinMode::Inner, equals_(_b_a, _c_a),
+        _stored_table_b,
+        _stored_table_c)),
+    _stored_table_a);
+
+  const auto lqp =
+  PredicateNode::make(greater_than_(_c_a, 150),
+    PredicateNode::make(greater_than_(_c_a, 100),
+      barrier_predicate_node));
+
+  // Increase the predicate's output count so that it becomes an actual barrier in the _push_down_traversal subroutine.
```

Comment:
Igitt, that is possible?

Reply:
Ading a node as output? Yes, that's possible apparently :b